src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -1803,10 +1808,50 @@ export async function executeSilently(session: IJupyterSession, code: string): P
         }
     };
     await request.done;
-    traceInfo(`Executing silently Code (completed) = ${code.substring(0, 100).splitLines().join('\\n')}`);
+
+    const codeForLogging = code.substring(0, 100).splitLines().join('\\n');
+
+    // Handle any errors logged in the output if needed
+    if (errorOptions) {
+        handleExecuteSilentErrors(outputs, errorOptions, codeForLogging);
+    }
+
+    traceInfo(`Executing silently Code (completed) = ${codeForLogging}`);
 
     return outputs;
 }
+
+function handleExecuteSilentErrors(
+    outputs: nbformat.IOutput[],
+    errorOptions: SilentExecutionErrorOptions,
+    codeForLogging: string
+) {
+    outputs
+        .filter((output) => {
+            return output.output_type === 'error';
+        })
+        .forEach((outputError) => {
+            const errorOutput = outputError as nbformat.IError;
+            const outputMessage = `${errorOutput.ename}: ${errorOutput.evalue} \n ${errorOutput.traceback
+                .map((line) => `    ${line}`)
+                .join('\n')}`;
+            const fullMessage = `${errorOptions.traceErrorsMessage || ''} ${codeForLogging} ${outputMessage}`;
+            if (errorOptions.traceErrors) {
+                traceError(fullMessage);
+            } else {
+                traceWarning(fullMessage);
+            }
+
+            // Send telemetry if requested, no traceback for PII
+            if (errorOptions.logTelemetryErrors) {
+                sendTelemetryEvent(Telemetry.KernelStartupCodeFailure, undefined, {
+                    ename: errorOutput.ename,
```

Comment:
Would be good to have some context, whether this is user code, startup code, or debugger variable request code, etc